WordPress/WordPressAllTimeWidget/AllTimeViewController.swift

```diff
@@ -1,30 +1,183 @@
 import UIKit
 import NotificationCenter
+import WordPressKit
 
 class AllTimeViewController: UIViewController {
 
     // MARK: - Properties
 
+    // TODO: For testing only. Remove when table added.
+    @IBOutlet private var visitors: UILabel!
+    @IBOutlet private var views: UILabel!
+    @IBOutlet private var posts: UILabel!
+    @IBOutlet private var best: UILabel!
+    @IBOutlet private var url: UILabel!
+    ////
+
+    private var statsValues: AllTimeWidgetStats? {
+        didSet {
+            updateStatsLabels()
+        }
+    }
+
+    private var siteUrl: String = Constants.noDataLabel
+
+    private var siteID: NSNumber?
+    private var timeZone: TimeZone?
+    private var oauthToken: String?
+    private var isConfigured = false
+
+    private let numberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        return formatter
+    }()
+
     private let tracks = Tracks(appGroupName: WPAppGroupName)
 
     // MARK: - View
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        retrieveSiteConfiguration()
     }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        loadSavedData()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        saveData()
+    }
+
 }
 
 // MARK: - Widget Updating
 
 extension AllTimeViewController: NCWidgetProviding {
 
     func widgetPerformUpdate(completionHandler: (@escaping (NCUpdateResult) -> Void)) {
+        retrieveSiteConfiguration()
+
+        if !isConfigured {
+            DDLogError("All Time Widget: Missing site ID, timeZone or oauth2Token")
+
+            DispatchQueue.main.async {
+                // TODO: reload table here
+            }
+
+            completionHandler(NCUpdateResult.failed)
+            return
+        }
+
         tracks.trackExtensionAccessed()
-        completionHandler(NCUpdateResult.newData)
+        fetchData()
```

Comment:
Should fetchData call the completionHandler on completion?

Reply:
Wow, I missed that neither widget is doing that. Well spotted!